include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -248,12 +248,21 @@ ORT_API_STATUS(OrtSetSessionGraphOptimizationLevel, _Inout_ OrtSessionOptions* o
                GraphOptimizationLevel graph_optimization_level);
 
 /**
- * How many threads in the session thread pool.
+ * Controls the size of the thread pool used to parallelize the execution of tasks within individual nodes (ops)
  * Set it to 0 to make onnxruntime run as single threaded.
- * \param session_thread_pool_size <0, let the runtime choose a default. =0, Don't create extra threads. 
+ * \param intra_op_thread_pool_size <0, let the runtime choose a default. =0, Don't create extra threads. 
  *                                 >0, create a thread pool with size of this value.
  */
-ORT_API_STATUS(OrtSetSessionThreadPoolSize, _Inout_ OrtSessionOptions* options, int session_thread_pool_size);
+ORT_API_STATUS(OrtSetIntraOpThreadPoolSize, _Inout_ OrtSessionOptions* options, int intra_op_thread_pool_size);
+
```

Comment:
Would it be simpler to just have an API in terms of the number of threads used rather than threadpool sizes? In ThreadPool::ParallelFor we use the current thread in conjunction with thread pool threads, so the number of threads that could potentially be used is actually threadpool size + 1. 

It feels a bit more intuitive to me to have an API called OrtSetIntraOpNumThreads where 0 means default and 1 means single threaded.

Reply:
Hi, I just changed the it, the plus one part.